### PR TITLE
Index Cleanup - Notification

### PR DESCRIPTION
**Pull request by djabarovgeorge**
### What change does this PR introduce?

Why? (Context)

We want to remove as many inefficient indexs and queries as possible.

### Why was this change needed?

Stop using single indexes on the mongoose key implementation
Add compound indexes for common queries and patterns using MongoDB index best practices
Remove unused indexes from MongoDB atlas after implementing the new compound indexes

### CODE REVIEW

File: libs/dal/src/repositories/notification/notification.schema.ts [hunk 1]
```diff
@@ -82,6 +82,78 @@ notificationSchema.virtual('jobs', {
...
+ * Path: libs/dal/src/repositories/notification/notification.repository.ts
+ *    Context: findBySubscriberId()
+ *        Query: find({_environmentId: environmentId,
+ *                    _subscriberId: subscriberId,});
+ *
+ */
+notificationSchema.index({
+  _subscriberId: 1,
+  _environmentId: 1,
+});
```
Comment (p-fernandez) [inline libs/dal/src/repositories/notification/notification.schema.ts:96]: Just a doubt, the same as in the other index cleanup PR. Is not this index included in the following one?

Comment (djabarovgeorge) [reply libs/dal/src/repositories/notification/notification.schema.ts:96]: They are including one another but the order matters and the prefix of the index make one redundant. 
for example 

{
  _subscriberId: 1,
  _environmentId: 1,
}

{
  transactionId: 1,
  _subscriberId: 1,
  _environmentId: 1,
}

are not the same, 
but the following example has the same prefix so we can remove the first one.

{
  _subscriberId: 1,
  _environmentId: 1,
}

{
  _subscriberId: 1,
  _environmentId: 1,
    transactionId: 1,
}

Hopefully, i got your question right and my answer helped 😅

Comment (p-fernandez) [reply libs/dal/src/repositories/notification/notification.schema.ts:96]: Yes, it does. I forgot that the order matters. Thanks for explaining it to me again!

**Review by p-fernandez — APPROVED** [review]
🌟

**Pull request merged**